### PR TITLE
roadmap update

### PR DESCRIPTION
**Pull request by mikebz**
This is a roadmap document update in order to map it closer to what is being delivered.  In addition to the major feature outline the document links out to functions, catalog, kpt Book chapters.

The intent is to be able to share this roadmap with beta testers.  We will be updating it more frequently now that the delivery and features are going to be much more defined.

### CODE REVIEW

File: docs/ROADMAP.md [hunk 1]
```diff
@@ -1,53 +1,89 @@
...
+
+Draft of the *v1* release slated for Summer 2021.
+
+### 1. Declarative function pipeline
+
+kpt has added a declarative way to customize and validate configuration.  
+This allows you to run several mutation and validation 
+functions in a pipeline alleviating the need to create shell scripts that do 
+the same thing.  Further information can be found in the 
+[declarative function execution]  section of the Kpt Book.
```
Comment (frankfarzan) [inline docs/ROADMAP.md:13]: nit: "The Kpt Book". Same below.

File: docs/ROADMAP.md [hunk 2]
```diff
@@ -1,53 +1,89 @@
...
+Setters used to be a special entity without sufficient differentiation from
+KRM functions. In kpt v1 setters become just another function and drastically
+simplify syntax.  Configuring 4 setters used to take 20 lines of yaml 
+and now takes 6.  Setters also get recursively applied to sub-packages by
+default.  For further information on the setter function please visit: 
+[apply-setters documentation]. 
+
+### 3. Resource merge
+
+kpt package updates now default to the resource merge strategy
```
Comment (mortent) [inline docs/ROADMAP.md:26]: Nit: We usually refer to it as `resource-merge`.

File: docs/ROADMAP.md [hunk 3]
```diff
@@ -1,53 +1,89 @@
...
+simplify syntax.  Configuring 4 setters used to take 20 lines of yaml 
+and now takes 6.  Setters also get recursively applied to sub-packages by
+default.  For further information on the setter function please visit: 
+[apply-setters documentation]. 
+
+### 3. Resource merge
+
+kpt package updates now default to the resource merge strategy 
+which allows you to edit configuration with an text editor of your choice 
+and still be able to rebase with upstream changes.
```
Comment (mortent) [inline docs/ROADMAP.md:28]: We should say `merge`, not `rebase`.

File: docs/ROADMAP.md [hunk 4]
```diff
@@ -1,53 +1,89 @@
...
+### 6. Function catalog
+
+All of the hydration and validation logic has been moved from the kpt binary 
+to functions allowing for flexibility and security.  This enables new 
+scenarios like limiting the customization and validation to a subset of 
+allowed functions.  The function catalog has received additional functions, 
+examples and help. Please visit the [function catalog] for further information.
+
+## Ongoing work
+Since this is a draft of the release notes you should be aware of the
```
Comment (mortent) [inline docs/ROADMAP.md:62]: This is the (relatively) short term plans. I think we should still consider maintaining a longer-term roadmap for the larger features we plan to add to kpt.

**Review by mortent — COMMENTED** [review]
The content looks good. I'm not sure if we should add this to the roadmap doc, as this is feels more like high-level release notes.

File: docs/ROADMAP.md [hunk 5]
```diff
@@ -1,53 +1,89 @@
...
+ongoing work.  At the time of writing the team is in [Milestone 3], and we
+are planning to complete the work in [Milestone 4].  For instance as of 6/1/21
+[out of place hydration] support is still in the works.  The exact dates
+and features are going to be highly dependent on feedback we get from beta
+testers.
```
Comment (frankfarzan) [inline docs/ROADMAP.md:67]: I think you want to distinguish v1 milestones and post-v1 milestones (1.1). The scope and the timelines for v1 milestones is more precise.

I suggest linking to list of all Milestones which includes 1.1:

https://github.com/GoogleContainerTools/kpt/milestones

File: docs/ROADMAP.md [hunk 6]
```diff
@@ -1,53 +1,89 @@
 # kpt roadmap for 2021
```
Comment (frankfarzan) [inline docs/ROADMAP.md:1]: Please also have this as part of the FAQ:

`What's the status and roadmap of kpt?`

File: docs/ROADMAP.md (hunk 2 above)
Comment (mikebz) [reply docs/ROADMAP.md:26]: fixed

File: docs/ROADMAP.md (hunk 3 above)
Comment (mikebz) [reply docs/ROADMAP.md:28]: merge and merge... I was trying to avoid that.  I will use "get updates"

File: docs/ROADMAP.md (hunk 4 above)
Comment (mikebz) [reply docs/ROADMAP.md:62]: Let's handle that in a separate PR.

File: docs/ROADMAP.md (hunk 1 above)
Comment (mikebz) [reply docs/ROADMAP.md:13]: all changed to "The Kpt Book"

File: docs/ROADMAP.md (hunk 5 above)
Comment (mikebz) [reply docs/ROADMAP.md:67]: updated

File: docs/ROADMAP.md (hunk 6 above)
Comment (mikebz) [reply docs/ROADMAP.md:1]: done

File: site/faq/README.md [hunk 7]
```diff
@@ -74,6 +74,11 @@ Yes. kpt's architecture facilitates the developing programs which may
 generate or modify configuration. See the [Using Functions] for how to
 compose multiple programs together.
 
+#### **What is the roadmap for kpt?**
```
Comment (frankfarzan) [inline site/faq/README.md:77]: I would move this near the topic. I would order the questions from most important/common to the least. Can be done in a seperate PR.

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**